File: apps/api/app/schemas/zotero.py
"""Zotero sync & import schemas."""

from pydantic import BaseModel, Field

from app.schemas.papers import PaperResponse


class ZoteroImportRequest(BaseModel):
    api_key: str | None = None
    user_id: str | None = None
    collection_id: str | None = None
    csl_json_content: str | None = Field(default=None, max_length=1_000_000)


class ZoteroImportResponse(BaseModel):
    total_imported: int
    papers: list[PaperResponse]
    skipped_count: int = 0
    message: str


class ZoteroSyncRequest(BaseModel):
    api_key: str
    user_id: str
    collection_id: str | None = None


class ZoteroSyncResponse(BaseModel):
    synced_items_count: int
    new_papers: list[PaperResponse]
    last_synced_version: int | None = None

File: apps/web/src/lib/api/zotero.ts
import { request } from './client';
import type { PaperDTO } from './papers';

export interface ZoteroImportResponseDTO {
  total_imported: number;
  papers: PaperDTO[];
  skipped_count: number;
  message: string;
}

export interface ZoteroSyncResponseDTO {
  synced_items_count: number;
  new_papers: PaperDTO[];
  last_synced_version: number;
}

export const zoteroApi = {
  import: (
    projectId: string,
    data: { api_key?: string; user_id?: string; collection_id?: string; csl_json_content?: string }
  ) =>
    request<ZoteroImportResponseDTO>(`/projects/${projectId}/zotero/import`, {
      method: 'POST',
      body: JSON.stringify(data),
    }),
  sync: (projectId: string, data: { api_key: string; user_id: string; collection_id?: string }) =>
    request<ZoteroSyncResponseDTO>(`/projects/${projectId}/zotero/sync`, {
      method: 'POST',
      body: JSON.stringify(data),
    }),
};
